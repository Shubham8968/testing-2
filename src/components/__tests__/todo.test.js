import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import Todo from "../todo";

afterEach(() => {
  cleanup();
});

test("Should render non-completed todo ", () => {
  const todo = { id: 1, title: "wash dishes", completed: false , };

  render(<Todo todo={todo} />);
  const todoElement = screen.getByTestId("todo-1");
  expect(todoElement).toBeInTheDocument();
  expect(todoElement).toHaveTextContent("wash dishes");

});

test("Should render completed todo  ", () => {
  const todo = { id: 2, title: "Make dishes", completed: true,  };

  render(<Todo todo={todo} />);
  const todoElement = screen.getByTestId("todo-2");
  expect(todoElement).toBeInTheDocument();
  expect(todoElement).toHaveTextContent("Make dishes");
 expect(todoElement).toContainHTML("<strike data-testid=todo-2 >Make dishes</strike>")
});

test("matches snapshot", () => {
  const todo = { id: 1, title: "wash dishes", completed: false, };
  const tree = renderer.create(<Todo todo={todo} />).toJSON();
 expect(tree).toMatchSnapshot();
});
