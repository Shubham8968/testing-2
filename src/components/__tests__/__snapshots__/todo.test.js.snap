// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<h1
  data-testid="todo-1"
>
  wash dishes
</h1>
`;
